stdlib/test/builtin/test_string.mojo

```diff
@@ -1082,6 +1073,78 @@ def test_indexing():
     assert_equal(a[2], "c")
 
 
+def test_string_iter():
+    var vs = String("123")
+
+    # Borrow immutably
+    fn conc(vs: String) -> String:
+        var c = String("")
+        for v in vs:
+            c += v
+        return c
+
+    assert_equal(123, atol(conc(vs)))
+
+    concat = String("")
+    for v in vs.__reversed__():
+        concat += v
+    assert_equal(321, atol(concat))
+
+    # TODO
```

Comment:
Is something blocking this?

Reply:
`UnsafePointer` does not have a `store` or `__setitem__` method

Reply:
Thanks, please describe this in the TODO.